pages/user/RegistrationHandler.inc.php

```diff
@@ -78,10 +78,19 @@ function registerUser($args, &$request) {
 
 		if ($regForm->validate()) {
 			$regForm->execute();
-			if (Config::getVar('email', 'require_validation')) {
-				// Send them home; they need to deal with the
-				// registration email.
-				$request->redirect(null, 'index');
+			if (!Validation::isLoggedIn()) {
+				if (Config::getVar('email', 'require_validation')) {
+					// Send them home; they need to deal with the
+					// registration email.
+					$request->redirect(null, 'index');
+					$this->setupTemplate($request, true);
+					$templateMgr =& TemplateManager::getManager();
+					$templateMgr->assign('pageTitle', 'user.register.emailValidation');
+					$templateMgr->assign('errorMsg', 'user.register.emailValidationDescription');
+					$templateMgr->assign('backLink', $request->url(null, 'login'));
+					$templateMgr->assign('backLinkLabel', 'user.login');
+					$templateMgr->display('common/error.tpl');
```

Comment:
I think you'll need a `return` statement after `$templateMgr->display(...)`. The `$request->redirect` function that used to be here takes care of preventing further processing, but `$templateMgr->display(...)` doesn't.